<a href="https://colab.research.google.com/github/markmaloba/W3_indpendent_project/blob/master/Moringa_Data_Science_Prep_W3_Independent_Project_2019_10_Mark_Maloba_Python_notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
#Import libraries


import pandas as pd
import numpy as np


In [2]:
#Import the 3 usage datasets

MTNDS1 = pd.read_csv('/content/Telcom_dataset.csv')
MTNDS1.head(5)

PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [3]:
MTNDS2 = pd.read_csv('/content/Telcom_dataset2.csv')
MTNDS2.head(5)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO

[5 rows x 10 columns]

In [4]:
MTNDS3 = pd.read_csv('/content/Telcom_dataset3.csv')
MTNDS3.head(5)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[5 rows x 10 columns]

In [8]:
#Renaming the PRODUCT column in the first dataset

MTNDS1 = MTNDS1.rename(columns={'PRODUTC': 'PRODUCT'})
MTNDS1.head(5)

PRODUCT  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [9]:
#Combining the dataframes

Combo = pd.concat([MTNDS1, MTNDS2, MTNDS3], ignore_index=True)
Combo.head(5)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  


CELLID     CELL_ID CELL_ON_SITE COUNTRY_A  ... PRODUCT SIET_ID     SITE_ID VALUE
0    NaN  /TJNe+Mmtu            1     19e2e  ...   Voice     NaN  /TJNe+Mmtu     0
1    NaN  /+cKZKIp41            0     19e2e  ...     sms     NaN  /+cKZKIp41     0
2    NaN  /u0FSD+ahi            0     19e2e  ...     sms     NaN  /u0FSD+ahi    25
3    NaN  +cNeJzsTp3            1     19e2e  ...   Voice     NaN  +cNeJzsTp3    56
4    NaN  051KhYDCpv            1     19e2e  ...   Voice     NaN  051KhYDCpv    14

[5 rows x 15 columns]

In [11]:
#Selecting only the columns we need
Combo2 = Combo[['PRODUCT', 'VALUE']]
Combo2.head(5)

#NOTE: The NaN values are there because the different CSVs have the same column labeled differently 

PRODUCT  VALUE
0   Voice      0
1     sms      0
2     sms     25
3   Voice     56
4   Voice     14

In [12]:
#Proof that Combo2 is now a combination of all 3
print(MTNDS1.shape)
print(MTNDS2.shape)
print(MTNDS3.shape)
print(Combo2.shape)

(5001, 10)
(5001, 10)
(5001, 10)
(15003, 2)


# **Now we can assess how much income we are getting from all services.**

In [13]:
Combo2_grouped = Combo2.groupby ('PRODUCT')['VALUE'].sum()
Combo2_grouped.sort_values(ascending=False)

PRODUCT
Voice    312593
sms       53427
data      17287
Name: VALUE, dtype: int64

Data is clearly lagging behind the other 2 services. It's bringing in 18 times less revenue than voice.


# Cell towers assessment

In [16]:
#Import dataset
CELLSdf = pd.read_csv('/content/cells_geo.csv', delimiter  = ";")
CELLSdf.head(5)

Unnamed: 0   VILLES      STATUS  ... AREA     CELL_ID   SITE_CODE
0           0   ADJAME  In Service  ...  NaN  fbb2e75af3  0bb492ef17
1           1   ADJAME  In Service  ...  NaN  1502501d78  0bb492ef17
2           2   ADJAME  In Service  ...  NaN  0c2e182a0c  0bb492ef17
3           3  ASSINIE  In Service  ...  NaN  69bdfc69fd  8a1e6ce31f
4           4  ASSINIE  In Service  ...  NaN  7a1029d2ab  8a1e6ce31f

[5 rows x 12 columns]

In [20]:
#Busiest cities usage

CELLSdf2 = CELLSdf.groupby('VILLES')
sort_CELLSdf = CELLSdf2.size()
sort_CELLSdf.sort_values(ascending=False)[:30]

VILLES
COCODY          354
YOPOUGON        295
ABOBO           255
MARCORY         141
TREICHVILLE     132
ADJAME          110
YAMOUSSOUKRO    104
PLATEAU         103
KOUMASSI         91
BOUAKE           65
PORT-BOUET       61
DALOA            59
GAGNOA           55
SAN-PEDRO        55
ATTECOUBE        46
BONGOUANOU       42
DIVO             39
KORHOGO          36
BINGERVILLE      30
SOUBRE           28
GRAND-BASSAM     27
GUIGLO           27
ANYAMA           27
MEADJI           25
MAN              24
DUEKOUE          21
ABENGOUROU       21
ADZOPE           21
BONDOUKOU        21
ABOISSO          18
dtype: int64

In [25]:
#Abidjan towers in service

len(CELLSdf[(CELLSdf['LOCALISATION']== 'ABIDJAN') & (CELLSdf['STATUS'] == 'In Service')])

1636

In [23]:
#Abidjan towers not in service

len(CELLSdf[(CELLSdf['LOCALISATION']== 'ABIDJAN') & (CELLSdf['STATUS'] != 'In Service')])

33

In [28]:
print (100*(33/1636))

2.0171149144254277


Only 2% of Abidjan towers are not in service. By our assessment, this not an alarming number. 

In [26]:
#Rest of Cote D'Ivoire towers in service

len(CELLSdf[(CELLSdf['LOCALISATION']== 'INTERIEUR') & (CELLSdf['STATUS'] == 'In Service')])

2271

In [27]:
#Rest of Cote D'Ivoire towers not in service

len(CELLSdf[(CELLSdf['LOCALISATION']== 'INTERIEUR') & (CELLSdf['STATUS'] != 'In Service')])

34

In [30]:
print (100*(34/2271))

1.4971378247468077


Only 1.5% of towers in the rest of Cote d'Ivoire are not in service. By our assessment, this not an alarming number.

In [31]:
print (2271+1636)
print (100*(1636/(2271+1636)))

3907
41.87356027642692


**Conclusion: Out of 3907 towers that are in service countrywide, 41% are in the capital, Abidjan. By our assessment, this represents possible oversaturation, and MTN should work to gain a foothold in other areas or 'villes'.**